File: intelligent-bank-risk-control-system/backend/src/main/java/com/system/domain/LoanApplication.java
package com.system.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("loan_application")
public class LoanApplication {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String applicationNo;
    private Long customerId;
    private Integer applyType;
    private BigDecimal applyAmount;
    private Integer applyTerm;
    private String applyPurpose;
    private Integer creditScore;
    private BigDecimal suggestedLimit;
    private String currentStatus;
    private Integer finalResult;
    private BigDecimal finalAmount;
    private Integer finalTerm;
    private String remark;
    private Long createBy;
    private LocalDateTime createTime;
    private Long updateBy;
    private LocalDateTime updateTime;
    private Integer isDeleted;
}

File: intelligent-bank-risk-control-system/frontend/src/api/types/loanApplication.ts
export interface LoanApprovalRecordItem {
  id: number
  reviewLevel: number
  reviewerName: string
  action: string
  comment: string | null
  createTime: string
}

export interface LoanApplication {
  id: number
  applicationNo: string
  applyType: number
  applyTypeName: string
  applyAmount: number
  applyTerm: number
  applyPurpose: string
  currentStatus: string
  currentStatusName: string
  finalResult: number | null
  finalResultName: string
  finalAmount: number | null
  finalTerm: number | null
  remark: string | null
  createTime: string
  updateTime: string | null
  approvalRecords: LoanApprovalRecordItem[]
}

export interface LoanApplicationCreateRequest {
  applyType: number
  applyAmount: number
  applyTerm?: number
  applyPurpose: string
}

export interface LoanApplicationReviewRequest {
  action: 'PASS' | 'REJECT'
  comment?: string
  finalAmount?: number
  finalTerm?: number
}
